src/Agent.Worker/Release/Artifacts/FileContainer/HttpRetryOnTimeoutOptions.cs

```diff
@@ -0,0 +1,129 @@
+        }
+
+        /// <summary>
+        /// Gets a singleton read-only instance of the default settings.
+        /// </summary>
+        public static HttpRetryOnTimeoutOptions Default
+        {
+            get
+            {
+                return s_defaultOptions.Value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the coefficient which exponentially increases the backoff starting at <see cref="MinBackoff" />.
+        /// </summary>
+        public TimeSpan BackoffCoefficient
+        {
+            get
+            {
+                return m_backoffCoefficient;
+            }
+            set
+            {
+                ThrowIfReadonly();
+                m_backoffCoefficient = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the minimum backoff interval to be used.
+        /// </summary>
+        public TimeSpan MinBackoff
+        {
+            get
+            {
+                return m_minBackoff;
+            }
+            set
+            {
+                ThrowIfReadonly();
+                m_minBackoff = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the maximum backoff interval to be used.
+        /// </summary>
+        public TimeSpan MaxBackoff
+        {
+            get
+            {
+                return m_maxBackoff;
+            }
+            set
+            {
+                ThrowIfReadonly();
+                m_maxBackoff = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the maximum number of retries allowed.
+        /// </summary>
+        public Int32 MaxRetries
+        {
+            get
+            {
+                return m_maxRetries;
+            }
+            set
+            {
+                ThrowIfReadonly();
+                m_maxRetries = value;
+            }
+        }
+
+        /// <summary>
+        /// Ensures that no further modifications may be made to the retry options.
+        /// </summary>
+        /// <returns>A read-only instance of the retry options</returns>
+        public HttpRetryOnTimeoutOptions MakeReadonly()
+        {
+            if (Interlocked.CompareExchange(ref m_isReadOnly, 1, 0) == 0)
+            {
+                // Make any lists read-only here.
+            }
+
+            return this;
+        }
+
+        /// <summary>
+        /// Throws an InvalidOperationException if this is marked as ReadOnly.
+        /// </summary>
+        private void ThrowIfReadonly()
+        {
+            if (m_isReadOnly > 0)
+            {
+                throw new InvalidOperationException();
+            }
+        }
+
+        private Int32 m_isReadOnly;
+        private Int32 m_maxRetries;
+        private TimeSpan m_minBackoff;
+        private TimeSpan m_maxBackoff;
+        private TimeSpan m_backoffCoefficient;
+        private static readonly TimeSpan s_minBackoff = TimeSpan.FromSeconds(10);
```

Comment:
moved to separate file, yes we will want them to be configurable but until we add that support we will use default values.